pkg/interface/chat/src/js/components/chat.js

```diff
@@ -14,288 +14,299 @@ import { deSig } from '/lib/util';
-         className={"pl4 pt2 bb b--gray4 b--gray1-d bg-gray0-d flex relative" +
-         "overflow-x-scroll overflow-x-auto-l overflow-x-auto-xl flex-shrink-0"}
-         style={{ height: 48 }}>
-         <SidebarSwitcher
-           sidebarShown={this.props.sidebarShown}
-           popout={this.props.popout}
-         />
-         <Link to={`/~chat/` + isinPopout + `room` + props.station}
-         className="pt2 white-d">
-           <h2
-             className={"dib f9 fw4 lh-solid v-top " +
-             ((title === props.station.substr(1)) ? "mono" : "")}
-             style={{ width: "max-content" }}>
-             {title}
-           </h2>
-         </Link>
-         <ChatTabBar
-           {...props}
-           station={props.station}
-           numPeers={group.length}
-           isOwner={deSig(props.match.params.ship) === window.ship}
-           popout={this.props.popout}
-           api={props.api}
-         />
-       </div>
-       <div
-         className="overflow-y-scroll bg-white bg-gray0-d pt3 pb2 flex flex-column-reverse"
-         style={{ height: "100%", resize: "vertical" }}
-         onScroll={this.onScroll}>
-         <div
-           ref={el => {
-             this.scrollElement = el;
-           }}></div>
-           { (
-               !(props.station in props.chatSynced) &&
-               (reversedMessages.length > 0)
-             ) ? (
-                 <ResubscribeElement
-                   api={props.api}
-                   host={props.match.params.ship}
-                   station={props.station} />
-               ) : (<div/>)
-           }
-           {reversedMessages}
-       </div>
-       <ChatInput
-         api={props.api}
-         numMsgs={lastMsgNum}
-         station={props.station}
-         owner={deSig(props.match.params.ship)}
-         ownerContact={ownerContact}
-         permissions={props.permissions}
-         envelopes={props.envelopes}
-         contacts={props.contacts}
-         placeholder="Message..."
-       />
-     </div>
-   );
- }
+  constructor(props) {
+    super(props);
+ 
+    this.state = {
+      numPages: 1,
+      scrollLocked: false
+    };
+ 
+    this.hasAskedForMessages = false;
+    this.onScroll = this.onScroll.bind(this);
+ 
+    this.updateReadInterval = setInterval(
+      this.updateReadNumber.bind(this),
+      1000
+    );
+  }
+ 
+  componentDidMount() {
+    this.updateReadNumber();
+    this.askForMessages();
+  }
+ 
+  componentWillUnmount() {
+    if (this.updateReadInterval) {
+      clearInterval(this.updateReadInterval);
+      this.updateReadInterval = null;
+    }
+  }
+ 
+  componentDidUpdate(prevProps, prevState) {
+    const { props, state } = this;
+ 
+    if (
+      prevProps.match.params.station !== props.match.params.station ||
+      prevProps.match.params.ship !== props.match.params.ship
+    ) {
+      this.hasAskedForMessages = false;
+      
+      if (props.envelopes.length < 100) {
+        this.askForMessages();
+      }
```

Comment:
idk if I'm reading this correctly, please confirm: for the "newly created chat case" is this just gonna be trying to load in scrollback on every update until we hit 100 messages?

Reply:
No, if you notice, this case is only being called upon switching chatrooms. So any time you switch chatrooms and the chatroom you’ve switched to has fewer than 100 messages, it will askForMessages()

Reply:
No, if you notice, this case is only being called upon switching chatrooms. So any time you switch chatrooms and the chatroom you’ve switched to has fewer than 100 messages, it will askForMessages()